### ISSUE TITLE
[Adjust Rule] SIM908: odd transform for self-assign

### ISSUE THREAD

**Issue by Skylion007** [MartinThoma/flake8-simplify#141]
## Desired change

* **Rule(s)**: SIM986, SIM987
* **Adjustment**: In which way should it/they be adjusted?

## Explanation
* Invalid transformation for t
## Example

This is an example where the mentioned rule(s) would currently be suboptimal:

```python
        if "state_dict" in cpt:
            cpt = cpt["state_dict"]
```

```python
        cpt = cpt.get("state_dict", cpt)
```
Is actually harder to read in this case.
